media-libs/gegl/gegl-0.4.8.ebuild

```diff
@@ -103,78 +102,57 @@ src_prepare() {
-	#
-	#  - v4l support does not work with our media-libs/libv4l-0.8.9,
-	#    upstream bug at https://bugzilla.gnome.org/show_bug.cgi?id=654675
-	#
-	#  - There are two checks for dot, one controllable by --with(out)-graphviz
-	#    which toggles HAVE_GRAPHVIZ that is not used anywhere.  Yes.
-	#
-	#  - mrg is not in tree and gexiv2 support only has effect when mrg support
-	#    is enabled
-	#
-	# So that's why USE="exif graphviz lua v4l" got resolved.  More at:
-	# https://bugs.gentoo.org/show_bug.cgi?id=451136
-	#
-	econf \
-		--disable-docs \
-		--disable-profile \
-		--disable-silent-rules \
-		--disable-workshop \
-		--program-suffix=-${SLOT} \
-		--with-gdk-pixbuf \
-		--with-pango \
-		--without-libspiro \
-		$(use_enable cpu_flags_x86_mmx mmx) \
-		$(use_enable cpu_flags_x86_sse sse) \
-		$(use_enable debug) \
-		$(use_with cairo) \
-		$(use_with cairo pangocairo) \
-		--without-exiv2 \
-		$(use_with ffmpeg libavformat) \
-		$(_use_with_both ffmpeg test gexiv2) \
-		--without-graphviz \
-		$(use_with jpeg2k jasper) \
-		$(use_with lcms) \
-		$(use_with lensfun) \
-		--without-lua \
-		--without-mrg \
-		$(use_with openexr) \
-		$(use_with raw libraw) \
-		$(use_with sdl) \
-		$(use_with svg librsvg) \
-		$(use_with tiff libtiff) \
-		$(use_with umfpack) \
-		$(use_with v4l libv4l) \
-		$(use_with v4l libv4l2) \
-		$(use_enable introspection) \
-		$(use_with vala) \
+	local myeconfargs=(
+		# disable documentation as the generating is bit automagic
+		#    if anyone wants to work on it just create bug with patch
+		--disable-docs
+		# never enable altering of CFLAGS via profile option
+		--disable-profile
+		--disable-silent-rules
+		#  - Parameter --disable-workshop disables any use of Lua, effectivly
+		--disable-workshop
+		--program-suffix=-${SLOT}
+		--with-gdk-pixbuf
+		--with-pango
+		#  - Parameter --with-exiv2 compiles a noinst-app only, no use
+		#    but needed during testing
+		--without-exiv2
+		#  - There are two checks for dot, one controllable by --with(out)-graphviz
+		#    which toggles HAVE_GRAPHVIZ that is not used anywhere.  Yes.
+		--without-graphviz
+		# libspiro: not in portage main tree
+		--without-libspiro
+		--without-lua
+		--without-mrg
+		$(use_enable cpu_flags_x86_mmx mmx)
+		$(use_enable cpu_flags_x86_sse sse)
+		$(use_enable debug)
+		$(use_enable introspection)
+		$(use_with cairo)
+		$(use_with cairo pangocairo)
+		$(use_with ffmpeg libavformat)
+		$(use_with jpeg2k jasper)
+		$(use_with lcms)
+		$(use_with lensfun)
+		$(use_with openexr)
+		$(use_with raw libraw)
+		$(use_with sdl)
+		$(use_with svg librsvg)
+		$(use_with tiff libtiff)
+		$(use_with umfpack)
+		#  - v4l support does not work with our media-libs/libv4l-0.8.9,
+		#    upstream bug at https://bugzilla.gnome.org/show_bug.cgi?id=654675
+		$(use_with v4l libv4l)
+		$(use_with v4l libv4l2)
+		$(use_with vala)
 		$(use_with webp)
+	)
+
+	use test && myeconfargs+=( $(use_with ffmpeg gexiv2) )
```

Comment:
That refactoring no longer passes `--without-gexiv2` for `USE=-test`. Is that intended?

Reply:
D'oh. Slight oversight. I'll fix that.